src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -111,6 +111,16 @@ StorageObjectStorage::StorageObjectStorage(
         && !configuration->isDataLakeConfiguration();
     const bool do_lazy_init = lazy_init && !need_resolve_columns_or_format && !need_resolve_sample_path;
 
+    if (!columns_.empty())
```

Comment:
I would prefer to use flag `is_table_function` in this condition together with `columns_`. Even if user provided all the columns we shouldn't create table in constructor.